tests/testthat/test-as_gtable.R

```diff
@@ -139,13 +139,41 @@ test_that("gtable widths are set appropriately", {
     as.numeric(test$widths)[4]
   )
 
+  # Test relative columns
+  # + absolute width
+  test <- tbl %>%
+    cols_width(y ~ pct(30)) %>%
+    tab_options(table.width = px(500)) %>%
+    as_gtable(text_grob = dummy_text)
+  test <- as.numeric(test$widths[2:3])
+  expect_equal(test / sum(test), c(0.7, 0.3), tolerance = 1e-6)
+
+  # + relative width
+  test <- tbl %>%
+    cols_width(y ~ pct(30)) %>%
+    tab_options(table.width = pct(50)) %>%
+    as_gtable(text_grob = dummy_text)
+
+  expect_equal(
+    test$widths[2:3],
+    grid::unit.pmax(grid::unit(c(100, 0), "pt"), grid::unit(c(0.35, 0.15), "npc"))
+  )
+
+  # + unspecified width
+  test <- tbl %>%
+    cols_width(y ~ pct(30)) %>%
+    as_gtable(text_grob = dummy_text)
+  test <- as.numeric(test$widths[2:3])
+  expect_equal(test / sum(test), c(0.7, 0.3), tolerance = 1e-6)
+  expect_equal(min(test), 100)
+
   test <- tbl %>%
     cols_width(x ~ pct(20), y ~ px(200)) %>%
     as_gtable(tbl, text_grob = dummy_text)
 
   expect_equal(
     as.character(test$widths),
-    c("0.5null", "0.2npc", "150.5625points", "0.5null")
+    c("0.5null", "100points", "150.5625points", "0.5null")
   )
```

Comment:
Below is to illustrate changes to this test.
The HTML has a tiny bit of whitespace to the left of 'x' and plenty to the left of 'y'.
``` r
library(gt)

tbl <- data.frame(x = 1, y = 1) |>
  gt() |>
  cols_width(x ~ pct(20), y ~ px(200))

gtExtras::gt_reprex_image(tbl)
```

<img src="https://i.imgur.com/PAJG11a.png" width="520" />

Currently, too much space is allocated to the left of 'x'
``` r
plot(tbl)
```

![](https://i.imgur.com/vPYYQdK.png)<!-- -->

With this PR, we more closely mirror the HTML output:
``` r
devtools::load_all("~/packages/gt/")
#> ℹ Loading gt
plot(tbl)
```

![](https://i.imgur.com/M476xuo.png)<!-- -->

<sup>Created on 2024-12-09 with [reprex v2.1.1](https://reprex.tidyverse.org)</sup>